src/algorithm/proj.rs

```diff
@@ -27,6 +28,41 @@ impl Reproject for PointArray {
     }
 }
 
+macro_rules! iter_geo_impl {
+    ($type:ty, $builder_type:ty, $push_func:ident) => {
+        impl<O: OffsetSizeTrait> Reproject for $type {
+            fn reproject(&self, proj: &Proj) -> Result<Self> {
+                let mut output_array = <$builder_type>::new();
+
+                for maybe_geom in self.iter_geo() {
+                    if let Some(mut geom) = maybe_geom {
+                        geom.transform(proj)?;
```

Comment:
In the future, we can additionally implement reprojection on the geoarrow coordinates directly (possible even copy on write) to avoid the conversion cost to geo objects!